packages/@webex/plugin-meetings/src/statsAnalyzer/mqaUtil.ts

```diff
@@ -4,135 +4,278 @@ import {mean, max} from 'lodash';
+  const lastFecPacketsDiscarded = getLastTotalValue('fecPacketsDiscarded');
+
+  const totalPacketsReceived = getTotalValue('totalPacketsReceived');
+  const packetsLost = getTotalValue('totalPacketsLost');
+  const totalBytesReceived = getTotalValue('totalBytesReceived');
+  const totalFecPacketsReceived = getTotalValue('fecPacketsReceived');
+  const totalFecPacketsDiscarded = getTotalValue('fecPacketsDiscarded');
+
+  audioReceiver.common.common.direction =
+    statsResults[Object.keys(statsResults).find((mediaType) => mediaType.includes(baseMediaType))]
+      ?.direction || 'inactive';
+  audioReceiver.common.common.isMain = !baseMediaType.includes('-share');
+  audioReceiver.common.transportType = statsResults.connectionType.local.transport;
+
+  // add rtpPacket info inside common as also for call analyzer
+  audioReceiver.common.rtpPackets = totalPacketsReceived - lastPacketsReceived;
+
+  // Hop by hop are numbers and not percentage so we compare on what we sent the last min
+  // collect the packets received for the last min
+  const totalPacketsLost = packetsLost - lastPacketsLost;
+  audioReceiver.common.mediaHopByHopLost = totalPacketsLost;
+  audioReceiver.common.rtpHopByHopLost = totalPacketsLost;
+
+  const fecRecovered =
+    totalFecPacketsReceived -
+    lastFecPacketsReceived -
+    (totalFecPacketsDiscarded - lastFecPacketsDiscarded);
+  audioReceiver.common.fecPackets = fecRecovered;
+
+  audioReceiver.common.rtpBitrate = ((totalBytesReceived - lastBytesReceived) * 8) / 60 || 0;
+};
+
+export const getAudioReceiverStreamMqa = ({
+  audioReceiverStream,
+  statsResults,
+  lastMqaDataSent,
+  mediaType,
+}) => {
   const sendrecvType = STATS.RECEIVE_DIRECTION;
 
-  const lastPacketsReceived = lastMqaDataSent[mediaType]?.[sendrecvType].totalPacketsReceived || 0;
-  const lastPacketsLost = lastMqaDataSent[mediaType]?.[sendrecvType].totalPacketsLost || 0;
   const lastPacketsDecoded = lastMqaDataSent[mediaType]?.[sendrecvType].totalSamplesDecoded || 0;
   const lastSamplesReceived = lastMqaDataSent[mediaType]?.[sendrecvType].totalSamplesReceived || 0;
   const lastConcealedSamples = lastMqaDataSent[mediaType]?.[sendrecvType].concealedSamples || 0;
   const lastBytesReceived = lastMqaDataSent[mediaType]?.[sendrecvType].totalBytesReceived || 0;
   const lastFecPacketsReceived = lastMqaDataSent[mediaType]?.[sendrecvType].fecPacketsReceived || 0;
   const lastFecPacketsDiscarded =
     lastMqaDataSent[mediaType]?.[sendrecvType].fecPacketsDiscarded || 0;
+  const lastPacketsReceived = lastMqaDataSent[mediaType]?.[sendrecvType].totalPacketsReceived || 0;
+  const lastPacketsLost = lastMqaDataSent[mediaType]?.[sendrecvType].totalPacketsLost || 0;
 
   const {csi} = statsResults[mediaType];
-  if (csi && !audioReceiver.streams[0].common.csi.includes(csi)) {
-    audioReceiver.streams[0].common.csi.push(csi);
+  if (csi && !audioReceiverStream.common.csi.includes(csi)) {
```

Comment:
perhaps a function which will extract the csi and add it to the stream? then it could be generic and used for video receiver stream as well?